packages/mobile/src/walletConnect/saga.ts

```diff
@@ -134,21 +134,36 @@ export function* acceptRequest({
     const wallet: UnlockableWallet = yield call(getWallet)
 
     let result: any
-
-    if (method === SupportedActions.eth_signTransaction) {
-      yield call(unlockAccount, account)
-      result = (yield call(wallet.signTransaction.bind(wallet), params)) as EncodedTransaction
-    } else {
-      throw new Error('Unsupported action')
+    // Default error
+    let error: ErrorType = WalletConnectErrors.GENERIC
+    try {
+      // If no `result` is set here, error is presumed
+      switch (method) {
+        case SupportedActions.eth_signTransaction:
+          yield call(unlockAccount, account)
+          result = (yield call(wallet.signTransaction.bind(wallet), params)) as EncodedTransaction
+          break
+        case SupportedActions.eth_signTypedData:
+          yield call(unlockAccount, account)
+          result = yield call(wallet.signTypedData.bind(wallet), account, JSON.parse(params[1]))
+          break
+        default:
+          error = WalletConnectErrors.JSONRPC_REQUEST_METHOD_UNSUPPORTED
+      }
+    } catch (e) {
+      Logger.debug(TAG + '@acceptRequest error obtaining result: ', e.message)
```

Comment:
I'd prefer we assign the error `WalletConnectErrors.GENERIC` here but no biggy

Reply:
Did it this way to make sure error is for sure set (i.e. no weird case of neither error nor result being set), but maybe this is less readable. Can change to that way!